### DISCUSSION TITLE
The `await updateSession()` with `@supabase/ssr` blocks route navigations in Next.js

### DISCUSSION THREAD

**Discussion by trojanyao** [supabase/supabase#30670] [General]
According to the official guide [Setting up Server-Side Auth for Next.js](https://supabase.com/docs/guides/auth/server-side/nextjs?queryGroups=router&router=app), in step 4, there is an `await updateSession()` in middleware. Since there's a request `await supabase.auth.getUser()` in the `updateSession`, every time I navigate to a route that matches the middleware `matcher`, there's a long time to wait — that seriously affects the user experience.

Is there a better solution to this problem?

<img width="1460" alt="image" src="https://github.com/user-attachments/assets/02c125e4-06b2-47e1-8f74-5ac961753010">

**Comment by alessiotortora** [comment]
On every request you update the session is also a problem. did you find a solution ? if you check the auth logs, there are a lot of request for auth/v1/request. 

Should we block update session ? if the session is this valid ? and do the redirecting and checks in the next middleware ?
